hicommonwealth/commonwealth
# Persist topic/title for thread drafts

## Link to Issue
Closes: #3634

## Description of Changes
- Added `useDraft` hook for flexibility to implement drafts outside of quill editor (e.g. topic/title of thread)
- Refactors old draft logic to use `useDraft` hook
- Updates `TopicSelector` to be a controlled component
- Adds `Cancel` button for on new thread page, which clears all fields

## Test Plan
- Go to New Thread page, set topic + title + body, then refresh page– confirm all fields are restored
- Click Cancel button, confirm that all fields are cleared, refresh page– confirm all fields are empty
- (Regression check): Go to a thread view page, type a comment without submitting, refresh page– confirm that comment text is restored
- (Regression check): Click Cancel button, confirm that text is cleared, refresh page– confirm text is empty
- (Regression check): Go to view topic page -> Change Topic– confirm that topic selector works

## Deployment Plan
N/A

## Other Considerations
With the new `useDraft` hook, we can add consistent client-side save/restore behavior to various parts of the app.
